cms/djangoapps/contentstore/asset_storage_handlers.py

```diff
@@ -88,6 +88,51 @@ def handle_assets(request, course_key_string=None, asset_key_string=None):
     return HttpResponseNotFound()
 
 
+def get_asset_usage_path(request, course_key, asset_key_string):
+    """
+    Get a list of units with ancestors that use given asset.
+    """
+    course_key = CourseKey.from_string(course_key)
+    if not has_course_author_access(request.user, course_key):
+        raise PermissionDenied()
+    asset_location = AssetKey.from_string(asset_key_string) if asset_key_string else None
+    store = modulestore()
+    usage_locations = []
+    static_path = StaticContent.get_static_path_from_location(asset_location)
+    verticals = store.get_items(
+        course_key,
+        qualifiers={
+            'category': 'vertical'
+        },
+    )
+    blocks = []
+
+    for vertical in verticals:
+        blocks.extend(vertical.get_children())
+
+    for block in blocks:
+        is_video_block = getattr(block, 'category', '') == 'video'
```

Comment:
nit: this is great work and well-written code. I only add this as a nit in case you want to get suggestions to further improve the code, but you can really skip this and leave it as is - everything looks great.

 Up to here, this function is extremely well understandable. If we wanted optimal readability, would it maybe help to for example keep the function short, and not nest if-statements much? One (of many) ways to do this would be to simply extract the following lines to something like `_get_usage_info_for_block`. The best thing about this is that anybody reading the code would not need to read all the following lines of code to see what the code does at a glance. And also that function would be potentially reusable.

This is very high-level, since the rest of the platform code has low readability standards and huge functions anyway, so I wouldn't worry about doing any changes here unless you feel particularly passionate about readability.